tt_eager/tt_dnn/op_library/topk/single_core/single_core_topk.cpp

```diff
@@ -39,39 +39,49 @@ operation::ProgramWithCallbacks single_core_topk_interleaved(const Tensor &input
     // for streaming in input
     uint32_t num_cb_unit = 2;
 
+
+    // Two tiles are loaded in for topk_local_sort at a time, and we double buffer to avoid stalls, so allocate four tiles of space
+    // TODO: In theory if we have enough memory we could allocate 2*Wt tiles to reduce stalls
     uint32_t input_cb_index = CB::c_in0;
-    tt_metal::CircularBufferConfig input_cb_config = tt_metal::CircularBufferConfig(num_cb_unit * value_tile_size, {{input_cb_index, input_cb_data_format}})
+    tt_metal::CircularBufferConfig input_cb_config = tt_metal::CircularBufferConfig(
+        num_cb_unit * 2 * value_tile_size, {{input_cb_index, input_cb_data_format}})
 		.set_page_size(input_cb_index, input_tile_size);
     auto cb_input_tensor = tt_metal::CreateCircularBuffer(program, core, input_cb_config);
 
-    // populate this as input is streamed
+    // Two tiles are loaded in for topk_local_sort at a time, and we double buffer to avoid stalls, so allocate four tiles of space
+    // This CB carries the indices that are created in the reader kernel
     uint32_t index_cb_index = CB::c_in1;
-    tt_metal::CircularBufferConfig index_input_intermed0_config = tt_metal::CircularBufferConfig(num_cb_unit * index_tile_size, {{index_cb_index, index_cb_data_format}})
+    tt_metal::CircularBufferConfig index_input_intermed0_config = tt_metal::CircularBufferConfig(
+        num_cb_unit * 2* index_tile_size, {{index_cb_index, index_cb_data_format}})
 		.set_page_size(index_cb_index, index_tile_size);
     auto cb_index_tensor = tt_metal::CreateCircularBuffer(program, core, index_input_intermed0_config);
```

Comment:
I need two tiles from each CB for my LLK at a time. I was thinking that preloading two while the previous two are being processed would be good since I can start the llk immediately in the next iteration.

Reply:
Changed the style though.